lib/galaxy/webapps/galaxy/api/biotools.py

```diff
@@ -0,0 +1,298 @@
+        general_dict["name"] = tool_name
+        clean_dict(general_dict)
+        return general_dict
+    return innerfunc
+
+
+def edam_to_uri(edam):
+    """
+    Build edam uri from edam data or edam format
+    :param edam: edam data or format
+    :return: edam uri
+    """
+    uri = "http://edamontology.org/{}".format(edam)
+    return uri
+
+
+def clean_list(jsonlist):
+    """
+    Recursive function in order to clean the json tool registry
+    :param jsonlist: List part of a json tool registry
+    :return: None
+    """
+    nullindexlist = []
+    for elem in range(len(jsonlist)):
+        if isinstance(jsonlist[elem], dict):
+            clean_dict(jsonlist[elem])
+        if isinstance(jsonlist[elem], list):
+            clean_list(jsonlist[elem])
+        if len(jsonlist[elem]) == 0:
+            nullindexlist.append(elem)
+    if nullindexlist:
+        nullindexlist.sort(reverse=True)
+        for i in nullindexlist:
+            jsonlist.pop(i)
+    return
+
+
+def clean_dict(jsondict):
+    """
+    Recursive function in order to clean the json tool registry
+    :param jsondict: Dictionary part of a json tool registry
+    :return: None
+    """
+    for sonkey, sonvalue in jsondict.items():
+        if sonvalue:
+            if isinstance(sonvalue, dict):
+                clean_dict(sonvalue)
+            if isinstance(sonvalue, list):
+                clean_list(sonvalue)
+
+        if not sonvalue:
+            del jsondict[sonkey]
+    return
+
+
+def build_tool_name(tool_id, tool_version):
+    """
+    Build the tool registry name regarding its long or short galaxy tool id
+    :param tool_id: Galaxy tool id
+    :param tool_version: Galaxy tool version
+    :return: tool registry name
+    :rtype: string
+    """
+    tbl = string.maketrans('.:/', '___')
+    # with a long id galaxy, xml version is already in the id
+    if tool_version in tool_id:
+        return str(tool_id).translate(tbl)
+    else:
+        return str("_".join([tool_id, tool_version])).translate(tbl)
+
+
+def format_description(description):
+    """
+    Test the first and last char of a description and replace them
+    with the format adapted to Elixir
+    :param description: Tool Galaxy description
+    :return: Formatted tool Galaxy description
+    :rtype: string
+    """
+    try:
+        size = len(description)
+        if description[size - 1] == '.':
+            return description[0].upper() + description[1:size]
+        else:
+            return description[0].upper() + description[1:size] + '.'
+    except IndexError:
+        raise  description
+
+
+def get_source_registry(tool_id):
+    """
+    Take a tool id and return the toolshed url if the id is a long id or a empty string if id is a short id
+    :param tool_id: Galaxy tool id
+    :return: toolshed url for a tool or empty string
+    :rtype: string
+    """
+    try:
+        return "/".join(tool_id.replace('repos', 'view', 1).split('/')[0:-2])
```

Comment:
Galaxy versions > 16.01 return a tool_shed_repository entry for toolshed tools that looks like this:
````
 "tool_shed_repository": {"owner": "devteam", "changeset_revision": "e359ce20fbd9", "name": "fastq_quality_boxplot", "tool_shed": "toolshed.g2.bx.psu.edu"}
```
This is probably better suited for constructing a link, given that a toolshed may well be reachable at <server_url/toolshed>